src/Logging/Logging.Console/src/ConsoleLogger.cs

```diff
@@ -73,53 +73,121 @@ public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId,
-            // category and event id
-            logBuilder.Append(_loglevelPadding);
-            logBuilder.Append(logName);
-            logBuilder.Append("[");
-            logBuilder.Append(eventId);
-            logBuilder.AppendLine("]");
-
-            // scope information
-            GetScopeInformation(logBuilder);
-
-            if (!string.IsNullOrEmpty(message))
+            if (Options.Format == ConsoleLoggerFormat.Default)
             {
-                // message
-                logBuilder.Append(_messagePadding);
+                // Example:
+                // INFO: ConsoleApp.Program[10]
+                //       Request received
+
+                var logLevelColors = GetLogLevelConsoleColors(logLevel);
+                var logLevelString = GetLogLevelString(logLevel);
+                // category and event id
+                logBuilder.Append(_loglevelPadding);
+                logBuilder.Append(logName);
+                logBuilder.Append("[");
+                logBuilder.Append(eventId);
+                logBuilder.AppendLine("]");
+
+                // scope information
+                GetScopeInformation(logBuilder, multiLine: true);
+
+                if (!string.IsNullOrEmpty(message))
+                {
+                    // message
+                    logBuilder.Append(_messagePadding);
+
+                    var len = logBuilder.Length;
+                    logBuilder.AppendLine(message);
+                    logBuilder.Replace(Environment.NewLine, _newLineWithMessagePadding, len, message.Length);
+                }
+
+                // Example:
+                // System.InvalidOperationException
+                //    at Namespace.Class.Function() in File:line X
+                if (exception != null)
+                {
+                    // exception message
+                    logBuilder.AppendLine(exception.ToString());
+                }
 
-                var len = logBuilder.Length;
-                logBuilder.AppendLine(message);
-                logBuilder.Replace(Environment.NewLine, _newLineWithMessagePadding, len, message.Length);
+                var timestampFormat = Options.TimestampFormat;
+                // Queue log message
+                _queueProcessor.EnqueueMessage(new LogMessageEntry(
+                    message: logBuilder.ToString(),
+                    timeStamp: timestampFormat != null ? DateTime.Now.ToString(timestampFormat) : null,
+                    levelString: logLevelString,
+                    levelBackground: logLevelColors.Background,
+                    levelForeground: logLevelColors.Foreground,
+                    messageColor: DefaultConsoleColor,
+                    logAsError: logLevel >= Options.LogToStandardErrorThreshold
+                ));
             }
+            else if (Options.Format == ConsoleLoggerFormat.Systemd)
+            {
+                // systemd reads messages from standard out line-by-line in a '<pri>message' format.
```

Comment:
Is there a reference doc you could link here?

Reply:
http://0pointer.de/blog/projects/journal-submit.html printf section is the best I found. It has the same info as the comment.